odoo/openupgrade/openupgrade_loading.py

```diff
@@ -157,6 +157,8 @@ def isrelated(model, k):
                     sorted([x[0] for x in v.selection]))
             else:
                 properties['selection_keys'] = 'function'
+        elif v.type == 'binary':
+            properties['attachment'] = str(getattr(v, "attachment", False))
```

Comment:
compare.py is not field type aware, but you are right in the sense that you may also add the key here, so that the default value is set to False (probably a better choice, too): https://github.com/OCA/OpenUpgrade/blob/10.0/odoo/addons/openupgrade_records/models/openupgrade_record.py#L39